recipes/blast/build.sh

```diff
@@ -35,102 +40,99 @@ cp -rf "${SRC_DIR}/RpsbProc/src/"* src/app/RpsbProc/
-# sse42
-#   Don't enable SSE 4.2 when optimizing.
-# pcre:
-#   Disable pcre (Perl regex).
+CONFIGURE_FLAGS="$CONFIGURE_FLAGS --with-bz2=$PREFIX"
+# --with(out)-krb5:
+#   Enable kerberos (needed on OSX).
+CONFIGURE_FLAGS="$CONFIGURE_FLAGS --without-krb5"
+# --with(out)-gnutls:
+#   Enable gnutls.
+CONFIGURE_FLAGS="$CONFIGURE_FLAGS --without-gnutls"
+# --with(out)-sse42
+#   Disable SSE 4.2 when optimizing.
+#   Many CPU's don't have this instruction set
+CONFIGURE_FLAGS="$CONFIGURE_FLAGS --without-sse42"
+
+# platform-specific flags
+if [[ "$(uname)" = "Linux" ]]; then
+	# --with(out)-64:
+	#   Compile in 64-bit mode instead of 32-bit.
+	CONFIGURE_FLAGS="$CONFIGURE_FLAGS --with-64"
+	# --with(out)-dll:
+	#   Use dynamic instead of static linking.
+	CONFIGURE_FLAGS="$CONFIGURE_FLAGS --with-dll"
+	# --with(out)-runpath:
+	#   Set runpath for installed $PREFIX location.
+	CONFIGURE_FLAGS="$CONFIGURE_FLAGS --with-runpath=$LIB_INSTALL_DIR"
+	# --with(out)-hard-runpath:
+	#   Hard-code runtime path, ignoring LD_LIBRARY_PATH
+	#   (disallow LD_LIBRARY_PATH override on Linux).
+	CONFIGURE_FLAGS="$CONFIGURE_FLAGS --with-hard-runpath"
+	# --with(out)-openmp:
+	#   Enable OpenMP extensions for all projects.
+	CONFIGURE_FLAGS="$CONFIGURE_FLAGS --with-openmp"
+else
+	# --with(out)-openmp:
+	#   Disable OpenMP extensions for all projects.
+	CONFIGURE_FLAGS="$CONFIGURE_FLAGS --without-openmp"
+	# --with(out)-pcre:
+	#   Enable pcre (Perl regex).
+	CONFIGURE_FLAGS="$CONFIGURE_FLAGS --without-pcre"
+	# --with(out)-gcrypt:
+	#   Disable gcrypt (needed on OSX).
+	CONFIGURE_FLAGS="$CONFIGURE_FLAGS --without-gcrypt"
+	# --with(out)-zstd:
+	#   Do not use Zstandard.
+	CONFIGURE_FLAGS="$CONFIGURE_FLAGS --without-zstd"
+fi
 
 # Fixes building on unix (linux and osx)
 export AR="${AR} rcs"
 
-if [[ $(uname) = Linux ]] ; then
-    ./configure \
-        --with-bin-release \
-        --with-64 \
-        --with-mt \
-        --with-dll \
-        --with-openmp \
-        --without-autodep \
-        --without-makefile-auto-update \
-        --with-flat-makefile \
-        --without-caution \
-        --without-lzo \
-        --with-hard-runpath \
-        --with-runpath=$LIB_INSTALL_DIR \
-        --without-debug \
-        --with-experimental=Int8GI \
-        --with-strip \
-        --with-vdb=$PREFIX \
-        --with-z=$PREFIX \
-        --with-bz2=$PREFIX \
-        --without-krb5 \
-        --without-gnutls \
-        --without-sse42
-else
-    ./configure \
-        --with-bin-release \
-        --with-mt \
-        --without-openmp \
-        --with-flat-makefile \
-        --without-lzo \
-        --without-zstd \
-        --without-debug \
-        --with-experimental=Int8GI \
-        --with-strip \
-        --with-vdb=$PREFIX \
-        --with-z=$PREFIX \
-        --with-bz2=$PREFIX \
-        --without-krb5 \
-        --without-gnutls \
-        --without-sse42 \
-        --without-gcrypt \
-        --without-pcre
-fi
+./configure $CONFIGURE_FLAGS
```

Comment:
An attempt at documenting the `./configure` flags more clearly. May need to add the "why" to the non-obvious ones.